### PR TITLE
Create example for icmpv6(ndp) router advertisement

### PR DESCRIPTION
**Pull request by scrollins**

### CODE REVIEW

File: examples/icmpv6-router-advertisement/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#[macro_use]
+extern crate netbricks;
+extern crate fallible_iterator;
```
Comment (zeeshanlakhani) [inline examples/icmpv6-router-advertisement/src/main.rs]: we don't need this right?

File: examples/icmpv6-router-advertisement/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        sched.add_task(pipeline).unwrap();
+    }
+}
+
+pub fn process_icmpv6(ipv6: Ipv6) -> Result<Option<Ipv6>> {
+    if let Ok(Icmpv6Message::RouterAdvertisement(advert)) = ipv6.parse_icmpv6() {
+        let mut iter = advert.options();
+        while let Ok(Some(option)) = iter.next() {
+            if let NdpOptions::SourceLinkLayerAddress(_option_type) = option {
+                //process options here
```
Comment (zeeshanlakhani) [inline examples/icmpv6-router-advertisement/src/main.rs]: maybe we should try and update an option here as a good example?

**Comment by zeeshanlakhani** [comment]
@scrollins *expect.out is empty here? That may be your issue. Also, for convention, probably best to name it expect.out like the others.

File: examples/icmpv6-router-advertisement/src/main.rs (hunk 1 above)
Comment (scrollins) [reply examples/icmpv6-router-advertisement/src/main.rs]: we need it to parse the options

**Comment by scrollins** [comment]
@zeeshanlakhani i renamed it to match the others but its still failing. 
```
root@ubuntu-bionic:/opt/netbricks/examples/icmpv6-router-advertisement# ./check.sh
Current Cargo Incremental Setting: CARGO_INCREMENTAL=0 
Current Rust Backtrace Setting: 1
name: netbricks, secondary: false, pool size: 2047, cache size: 32
primary core: 0, cores: [0], strict: false
ports:
	name: dpdk:eth_pcap0,rx_pcap=data/icmpv6_router_advertisement.pcap,tx_pcap=/tmp/out.pcap, rxq: [0], txq: [0], rxd: 128, txd: 128, loopback: false, tso: false, csumoffload: false
DPDK args: None, duration: Some(1)
Failed to detect # of NUMA nodes from: /sys/devices/system/node/possible. Assuming a single-node system...
EAL: Detected 4 lcore(s)
EAL: Detected 1 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/netbricks/mp_socket
EAL: Probing VFIO support...
PMD: pmd port 0 configured successfully
Receiving started
Running 1 pipelines
reading from file /tmp/out.pcap, link-type EN10MB (Ethernet)
0a1,3
> 00:00:00:00:00:00 > 00:00:00:00:00:01, ethertype IPv6 (0x86dd), length 64: fe80::1.1025 > fe80::2.1024: Flags [S], seq 1, win 10, length 0
> 
> 
----
FAIL```

File: examples/icmpv6-router-advertisement/src/main.rs (hunk 2 above)
Comment (scrollins) [reply examples/icmpv6-router-advertisement/src/main.rs]: i'll work on an example to update an option

**Comment by zeeshanlakhani** [comment]
@drunkirishcoder makes a good point to show the complete example. There's a timer in the spec for how often to send a solicitation, and then we can process adverts sent out iand store that in a hashmap.

**Pull request closed without merging**